### PR TITLE
26424 Credit Memo Misc Items

### PR DESCRIPTION
**Pull request by andersdd**
Add miscellaneous (non-inventory) items to Sales Credit Memos.  Also complete alternate revenue account per item development

### CODE REVIEW

**Comment by gilmoskowitz** [comment]
I think this is failing CI with the following:
```
2015-09-04 23:33:46 UTC WARNING:  Error: relation "cmitemaccnt" does not exist
	    at Object.deleteRecord (eval at <anonymous> (js_init:921:20))
	    at Object.deleteRecord (eval at <anonymous> (js_init:921:20))
	    at Object.commitRecord (eval at <anonymous> (js_init:921:20))
	    at Object.delete (eval at <anonymous> (js_init:921:20))
	    at delete:7:26
	    at [object Object].<anonymous> (delete:13:2)
```

File: foundation-database/public/functions/postcreditmemo.sql [hunk 1]
```diff
@@ -192,6 +195,86 @@ BEGIN
...
+                                 getPrjAccntId(_p.cmhead_prj_id, 
+                                 COALESCE(_r.cmitem_rev_accnt_id,_r.salescat_sales_accnt_id)), 
+                                 round(currToBase(_p.cmhead_curr_id,
+                                                _r.extprice * -1,
+                                                _p.cmhead_docdate), 2),
+                                 _glDate, _p.cmhead_billtoname ) INTO _test;
+      IF (_test < 0) THEN
+        PERFORM deleteGLSeries(_sequence);
+        raise exception 'made it here';
+        RETURN -12;
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/postcreditmemo.sql]: Convert this to
```
RAISE EXCEPTION 'Could not debit the sales account for the credit memo line item [xtuple: postcreditmemo, -12, %1, %2]',
      _p.cmhead_number, _r.cmitem_itemsite_id;
```
or something similar.

Comment (andersdd) [inline foundation-database/public/functions/postcreditmemo.sql]: Woops you spotted my deliberate error msg  :-)

**Comment by andersdd** [comment]
cmitemaccnt is the unnecessary table I removed.  I will look further into what might still be calling it.

**Pull request merged**